graylog2-server/src/main/java/org/graylog2/indexer/fieldtypes/IndexFieldTypePollerPeriodical.java

```diff
@@ -0,0 +1,273 @@
+                    .forEach(types -> dbService.delete(types.id()));
+        });
+    }
+
+    /**
+     * Creates a new field type polling job for the newly created index set.
+     * @param event index set creation event
+     */
+    @Subscribe
+    public void handleIndexSetCreation(final IndexSetCreatedEvent event) {
+        final String indexSetId = event.indexSet().id();
+        // We are NOT using IndexSetRegistry#get(String) here because of this: https://github.com/Graylog2/graylog2-server/issues/4625
+        final Optional<IndexSetConfig> optionalIndexSet = indexSetService.get(indexSetId);
+
+        if (optionalIndexSet.isPresent()) {
+            schedule(mongoIndexSetFactory.create(optionalIndexSet.get()));
+        } else {
+            LOG.warn("Couldn't find newly created index set <{}>", indexSetId);
+        }
+    }
+
+    /**
+     * Removes the field type polling job for the now deleted index set.
+     * @param event index set deletion event
+     */
+    @Subscribe
+    public void handleIndexSetDeletion(final IndexSetDeletedEvent event) {
+        final String indexSetId = event.id();
+
+        LOG.debug("Disable field type updating for index set <{}>", indexSetId);
+        cancel(futures.remove(indexSetId));
+    }
+
+    /**
+     * Removes the index field type data for the deleted index.
+     * @param event index deletion event
+     */
+    @Subscribe
+    public void handleIndexDeletion(final IndicesDeletedEvent event) {
+        event.indices().forEach(indexName -> {
+            LOG.debug("Removing field type information for deleted index <{}>", indexName);
+            dbService.delete(indexName);
+        });
+    }
+
+    /**
+     * Creates a new polling job for the given index set to keep the active write index information up to date.
+     * @param indexSet index set
+     */
+    private void schedule(final IndexSet indexSet) {
+        final String indexSetId = indexSet.getConfig().id();
+        final String indexSetTitle = indexSet.getConfig().title();
+        final Duration refreshInterval = indexSet.getConfig().fieldTypeRefreshInterval();
+
+        if (Duration.ZERO.equals(refreshInterval)) {
+            LOG.debug("Skipping index set with ZERO refresh interval <{}/{}>", indexSetTitle, indexSetId);
+            return;
+        }
+        if (!indexSet.getConfig().isWritable()) {
+            LOG.debug("Skipping non-writable index set <{}/{}>", indexSetTitle, indexSetId);
+            return;
+        }
+
+        // Make sure there is no existing polling job running for this index set
+        cancel(futures.get(indexSetId));
+
+        LOG.debug("Schedule index field type updating for index set <{}/{}> every {} ms", indexSetId, indexSetTitle,
+                refreshInterval.getMillis());
+        final ScheduledFuture<?> future = scheduler.scheduleAtFixedRate(() -> {
+            try {
+                // Only check the active write index on a regular basis, the others don't change anymore
```

Comment:
What happens if the deflector cycles quicker than our refresh interval?
Could we miss one index in that case?

Reply:
We calculate the types on index rotation in `SetIndexReadOnlyAndCalculateRangeJob`. That should catch all indices.

![image](https://user-images.githubusercontent.com/461/37295690-14ff8c4a-2619-11e8-9669-4e54325ff046.png)